core-web/libs/edit-content/src/lib/fields/dot-edit-content-category-field/components/dot-category-field-search-list/dot-category-field-search-list.component.ts

```diff
@@ -0,0 +1,177 @@
+import { Subject } from 'rxjs';
+
+import { CommonModule } from '@angular/common';
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    Component,
+    computed,
+    DestroyRef,
+    effect,
+    ElementRef,
+    EventEmitter,
+    inject,
+    input,
+    Output,
+    signal,
+    ViewChild
+} from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+import { SkeletonModule } from 'primeng/skeleton';
+import { TableModule } from 'primeng/table';
+
+import { debounceTime } from 'rxjs/operators';
+
+import { DotMessagePipe } from '@dotcms/ui';
+
+import {
+    DotCategoryFieldCategory,
+    DotCategoryFieldKeyValueObj,
+    DotTableHeaderCheckboxSelectEvent,
+    DotTableRowSelectEvent
+} from '../../models/dot-category-field.models';
+import { getParentPath } from '../../utils/category-field.utils';
+import { DotTableSkeletonComponent } from '../dot-table-skeleton/dot-table-skeleton.component';
+
+const DELAY_FOR_LISTENER = 300;
+
+@Component({
+    selector: 'dot-category-field-search-list',
+    standalone: true,
+    imports: [CommonModule, TableModule, SkeletonModule, DotTableSkeletonComponent, DotMessagePipe],
+    templateUrl: './dot-category-field-search-list.component.html',
+    styleUrl: './dot-category-field-search-list.component.scss',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DotCategoryFieldSearchListComponent implements AfterViewInit {
+    /**
+     * Represents a reference to a table container element in the DOM to calculate the
+     * viewport to use in the virtual scroll
+     */
+    @ViewChild('tableContainer', { static: false }) tableContainer!: ElementRef;
+
+    /**
+     * The scrollHeight variable represents a signal with a default value of '0px'.
+     * It can be used to track and manipulate the height of a scrollable element.
+     */
+    scrollHeight = signal<string>('0px');
```

Comment:
@oidacra According to the conversations in Slack, all signals should start with `$`.

Reply:
Maybe it should be a const like DEFAULT_SCROLL_HEIGHT = '0px' ?